metrics/performance/first-meaningful-paint.js

```diff
@@ -18,49 +18,61 @@
 
 const DevtoolsTimelineModel = require('devtools-timeline-model');
 
+const FAILURE_ERROR = new Error('Navigation and first paint timings not found');
+
 class FirstMeaningfulPaint {
+
   /**
    * @param {!Array<!Object>} traceData
    */
   static parse(traceData) {
     return new Promise((resolve, reject) => {
+      if (!traceData || !Array.isArray(traceData)) {
+        return reject(FAILURE_ERROR);
```

Comment:
you can also just `throw new Error(FAILURE_MESSAGE)`. Not sure if the difference is semantically meaningful, though